src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py

```diff
@@ -49,6 +49,85 @@ def test_parse_asn1_date(self):
+                 checks=self.check('length(newVNet.subnets)', 1))
+        self.cmd('network vnet subnet update -n {subnet} --vnet-name {vnet} -g {rg} '
+                 '--disable-private-endpoint-network-policies true',
+                 checks=self.check('privateEndpointNetworkPolicies', 'Disabled'))
+
+        # Create a private endpoint connection
+        pe = self.cmd('network private-endpoint create -g {rg} -n {pe} --vnet-name {vnet} --subnet {subnet} -l {loc} '
+                      '--connection-name {pe_connection} --private-connection-resource-id {kv_id} '
+                      '--group-ids vault').get_output_in_json()
+        self.kwargs['pe_id'] = pe['id']
+
+        # Show the connection at vault side
+        keyvault = self.cmd('keyvault show -n {kv}',
+                            checks=self.check('length(properties.privateEndpointConnections)', 1)).get_output_in_json()
+        self.kwargs['kv_pe_id'] = keyvault['properties']['privateEndpointConnections'][0]['id']
+        self.cmd('keyvault private-endpoint show --connection-id {kv_pe_id}',
+                 checks=self.check('id', '{kv_pe_id}'))
+        self.kwargs['kv_pe_name'] = self.kwargs['kv_pe_id'].split('/')[-1]
+        self.cmd('keyvault private-endpoint show -n {kv} --connection-name {kv_pe_name}',
+                 checks=self.check('name', '{kv_pe_name}'))
+
+        # Try running `set-policy` on the linked vault
+        self.kwargs['policy_id'] = keyvault['properties']['accessPolicies'][0]['objectId']
+        self.cmd('keyvault set-policy -g {rg} -n {kv} --object-id {policy_id} --certificate-permissions get list',
+                 checks=self.check('length(properties.accessPolicies[0].permissions.certificates)', 2))
+
+        # Test approval/rejection
+        self.kwargs.update({
+            'approval_desc': 'You are approved!',
+            'rejection_desc': 'You are rejected!'
+        })
+        self.cmd('keyvault private-endpoint reject --connection-id {kv_pe_id} '
+                 '--rejection-description "{rejection_desc}"', checks=[
+                     self.check('privateLinkServiceConnectionState.status', 'Rejected'),
+                     self.check('privateLinkServiceConnectionState.description', '{rejection_desc}')
+                 ])
+
+        max_retries = 20
+        retries = 0
+        while self.cmd('keyvault private-endpoint show --connection-id {kv_pe_id}').\
+                get_output_in_json()['provisioningState'] != 'Succeeded' or retries > max_retries:
+            time.sleep(5)
+            retries += 1
+
+        self.cmd('keyvault private-endpoint approve -n {kv} --connection-name {kv_pe_name} '
+                 '--approval-description "{approval_desc}"', checks=[
+                     self.check('privateLinkServiceConnectionState.status', 'Approved'),
+                     self.check('privateLinkServiceConnectionState.description', '{approval_desc}')
+                 ])
```

Comment:
Let's check `provisioningState` in the return value as well.

Reply:
Sure, will add this.